homeassistant/components/glances/sensor.py

```diff
@@ -38,192 +37,192 @@ class GlancesSensorEntityDescription(SensorEntityDescription):
     name_suffix: str | None = None
 
 
-SENSOR_TYPES: tuple[GlancesSensorEntityDescription, ...] = (
-    GlancesSensorEntityDescription(
+SENSOR_TYPES: dict[tuple[str, str], GlancesSensorEntityDescription] = {
```

Comment:
Could this change to `SENSOR_TYPES` also be implemented as a preliminary PR or as a follow-up PR to the dependency bump?

Reply:
The change to the SENSOR_TYPES is linked to the new method introduced in the updated library. I am trying to simplify the code for adding new sensor entities under `async_setup_entry`. And also the same for updating the native value of the sensors.
The change is merely identifying each description by (type, key). No other changes are done for the description itself.

Reply:
I still feel that it could (should?) be a separate PR.

Reply:
I will try and check.

Reply:
Keeping SENSOR_TYPES as they are now will make looping through them under `async_setup_entry` more difficult considering the new structure of the data returned from the updated library. Let me show you an example.
Sample data returned from library:
```json
{
    "fs": {
        "/ssl": {"disk_use": 30.7, "disk_use_percent": 6.7, "disk_free": 426.5},
        "/media": {"disk_use": 30.7, "disk_use_percent": 6.7, "disk_free": 426.5},
    },
    "sensors": {"cpu_thermal 1": {"temperature_core": 59}},
    "mem": {
        "memory_use_percent": 27.6,
        "memory_use": 1047.1,
        "memory_free": 2745.0,
    },
    "docker": {"docker_active": 2, "docker_cpu_use": 77.2, "docker_memory_use": 1149.6},
}
```
Looping through the data (instead of looping through SENSOR_TYPES) when adding the sensors allows us to create sensors for each sensor key under each label for each sensor type. To select the correct SENSOR_TYPE we need to identify it by (type, key). That would be the best way to do so.

Reply:
Yes, but the "old" way still works in the new glances package 0.4.1. Or it as has been removed?
So it would be better to have:
- one PR for dependency bump (no code changes)
- one follow-up PR for migrating to use the new `get_ha_sensor_data` method.

Reply:
Yes it still works. I can separate the PRs as suggested.